SYCL/ESIMD/api/hardware_dispatch.cpp

```diff
@@ -0,0 +1,81 @@
+//==---------------- hardware_dispatch.cpp  - ESIMD hardware dispatch test -==//
```

Comment:
I believe 'api' filder is mostly for ESIMD specific API/functions.
The functionality being tested in this test is more common. Probably, it is better to just place the test to SYCL/ESIMD